scenario/consistency_checker.py

```diff
@@ -92,6 +93,27 @@ def check_consistency(
         )
 
 
+def check_resource_consistency(
+    *,
+    state: "State",
+    charm_spec: "_CharmSpec",
+    **_kwargs,  # noqa: U101
+) -> Results:
+    """Check the internal consistency of the resources from metadata and in State."""
+    errors = []
+    warnings = []
+
+    resources_from_meta = set(charm_spec.meta.get("resources", {}).keys())
+    resources_from_state = set(state.resources.keys())
```

Comment:
I believe you can drop the `.keys()` because the default iterator for dicts is the keys.